src/map/ai/helpers/pathfind.cpp

```diff
@@ -179,23 +176,24 @@ bool CPathFind::PathInRange(const position_t& point, float range, uint8 pathFlag
             }
             else
             {
-                return false;
+                result = false;
             }
         }
-        else if (point.y - m_POwner->loc.p.y > 4) // 4 Is generally a decent sized ledge.
+        else // I'm probably stuck on a rock or something dumb.
         {
-            if (PMob->isInDynamis() && PMob->PAI->IsEngaged()) // Ignore wallhack for dynamis.
-            {
-                PMob->PAI->Disengage();
-            }
-            else
-            {
-                result = false; // Make me fall down.
-            }
+            result = false; // Make me go down or up.
         }
-        else if (point.y - m_POwner->loc.p.y > 1) // I'm probably stuck on a rock or something dumb.
+    }
+
+    if (m_POwner->objtype == TYPE_MOB &&
+        m_POwner->loc.zone->m_updatedNavmesh)
+    {
+        auto PEntity = dynamic_cast<CBattleEntity*>(m_POwner)->GetBattleTarget();
+
+        if (PEntity)
         {
-            result = false; // Make me go down.
+            result           = abs(m_POwner->loc.p.y - PEntity->loc.p.y) < 0.25 ? ValidPosition(PEntity->loc.p) : true;
```

Comment:
We should probably getVerticalLimit here

Reply:
Made the function and set it up.